pandas/tests/groupby/test_groupby.py

```diff
@@ -2037,3 +2037,16 @@ def test_groupby_list_level():
     expected = pd.DataFrame(np.arange(0, 9).reshape(3, 3))
     result = expected.groupby(level=[0]).mean()
     tm.assert_frame_equal(result, expected)
+
+
+def test_groups_repr_truncates():
+    # GH 1135
+    df = pd.DataFrame({"a": [1, 1, 1, 2, 2, 3], "b": [1, 2, 3, 4, 5, 6]})
+
+    with pd.option_context("display.max_rows", 2):
+        x = df.groupby("a").groups
```

Comment:
This is giving `{1: [0, 1, ...], 2: [3, 4], ...}` so altering max_seq_items is not ideal. Is there another way?

Reply:
@simonjayhawkins could test this on a longer DataFrame and then not alter `display.max_rows`, would that be OK?

Reply:
use `with pd.option_context("display.max_seq_items", 2):` and just `return pprint_thing(self)`  from `repr` instead?

Reply:
Thanks for your review - I've reread this sentence a few times

>  just return pprint_thing(self) from repr instead

but I don't understand what you mean. If I change the definition of `PrettyDict.__repr__` to
```
        return pprint_thing(self)
```
then won't the issue be not-solved? At least, with the test as it's currently written, it fails

Reply:
I thought pprint_thing called_pprint_dict which uses display.max_seq_items as the default. I'll check in a bit.

Reply:
> I thought pprint_thing called_pprint_dict which uses display.max_seq_items as the default. I'll check in a bit.

confirmed.

> then won't the issue be not-solved?

if the top level of the dict **must** follow display.max_rows for truncation, then maybe updating pprint_thing to accept, say, a tuple for max_seq_items to allow the nested sequences to follow display.max_seq_items could be an option. The tuple could be any length that specifies the max_seq_items for each level with the last value being used for deeper sequences.

so something like `return pprint_thing(self, max_seq_items=(get_option("display.max_rows"), get_option("display.max_seq_items"))` ?

or maybe alter the \_\_repr__ to be less dict-like

Reply:
> At least, with the test as it's currently written, it fails

sorry, my mistake here, hadn't realised the default for max_seq_items is longer than that for max_rows, I should've checked before commenting. Have updated according to your original request